File: backend/src/main/java/com/example/demo/entity/Image.java
package com.example.demo.entity;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.util.Date;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "images")
public class Image {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String imageUrl;

    private Date createdAt;

    private String createdBy;

    @ManyToOne()
    @JoinColumn(name = "product_color_id")
    @JsonIgnore
    private ProductColor productColor;
}

File: frontend/types/image.ts
export type UploadImageProps = {
  size: number;
  imageUrl?: string;
  imagesList: any[];
  addImage: (newImageUrl: string) => void;
  removeImage: (newImageUrl: string) => void;
};

export type Image = {
  id?: number;
  createdAt?: Date;
  lastModifiedAt?: Date;
  createdBy?: string;
  imageUrl: string;
  // productColor?: ProductColo
};
